Sources/Operators.swift

```diff
@@ -0,0 +1,181 @@
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import CoreGraphics
+import Foundation
+import UIKit
+
+
+// MARK: – ViewPosition Operators
+
+
+infix operator --> : AssignmentPrecedence
+infix operator <|> : AdditionPrecedence
+
+/// Aligns lhs to rhs.
+/// - parameter lhs: The ViewPosition to align.
+/// - parameter rhs: The position to which lhs is aligned.
+public func -->(lhs: ViewPosition, rhs: ViewPosition) {
+    lhs.align(to: rhs)
+}
+
+/// Aligns lhs to the rhs's position on the lhs's superview.
+/// - parameter lhs: The ViewPosition to align.
+/// - parameter rhs: The position on the lhs's superview to which lhs is aligned.
+public func -->(lhs: ViewPosition, rhs: ViewPosition.Anchor) {
+    lhs.align(toSuperviewAnchor: rhs)
+}
+
+/// Aligns lhs to the rhs's position on the lhs's superview.
+/// - parameter lhs: The ViewPosition to align.
+/// - parameter rhs: The position on the lhs's superview to which lhs is aligned.
+public func -->(lhs: ViewPosition, rhs: ViewPosition.OffsetAnchor) {
+    lhs.align(toSuperviewAnchor: rhs.anchor, xOffset: rhs.offset.horizontal, yOffset: rhs.offset.vertical)
+}
+
+public func +(lhs: ViewPosition, rhs: UIOffset) -> ViewPosition {
+    return ViewPosition(view: lhs.view, position: CGPoint(x: lhs.anchor.x + rhs.horizontal, y: lhs.anchor.y + rhs.vertical))
+}
+
+public func -(lhs: ViewPosition, rhs: UIOffset) -> ViewPosition {
+    return ViewPosition(view: lhs.view, position: CGPoint(x: lhs.anchor.x - rhs.horizontal, y: lhs.anchor.y - rhs.vertical))
+}
+
+public func +(lhs: ViewPosition.Anchor, rhs: UIOffset) -> ViewPosition.OffsetAnchor {
+    return ViewPosition.OffsetAnchor(offset: rhs, anchor: lhs)
+}
+
+public func -(lhs: ViewPosition.Anchor, rhs: UIOffset) -> ViewPosition.OffsetAnchor {
+    return ViewPosition.OffsetAnchor(offset: rhs, anchor: lhs)
+}
+
+public func +(lhs: ViewPosition.OffsetAnchor, rhs: UIOffset) -> ViewPosition.OffsetAnchor {
+    return ViewPosition.OffsetAnchor(offset: lhs.offset + rhs, anchor: lhs.anchor)
+}
+
+public func -(lhs: ViewPosition.OffsetAnchor, rhs: UIOffset) -> ViewPosition.OffsetAnchor {
+    return ViewPosition.OffsetAnchor(offset: lhs.offset - rhs, anchor: lhs.anchor)
+}
+
+
+// MARK: - DistributionItem Operators
+
+
+public func <|>(lhs: DistributionItem, rhs: DistributionItem) -> [DistributionItem] {
+    return [lhs, rhs]
+}
+
+public func <|>(lhs: [DistributionItem], rhs: DistributionItem) -> [DistributionItem] {
+    var bothSides = lhs
+    bothSides.append(rhs)
+    return bothSides
+}
+
+public func <|>(lhs: [DistributionItem], rhs: UIView) -> [DistributionItem] {
+    return lhs <|> .view(rhs)
+}
+
+public func <|>(lhs: UIView, rhs: DistributionItem) -> [DistributionItem] {
+    return .view(lhs) <|> rhs
+}
+
+public func <|>(lhs: DistributionItem, rhs: UIView) -> [DistributionItem] {
+    return lhs <|> .view(rhs)
+}
+
```

Comment:
So, right now the DSL for distribution is... kinda weak. Basically just a concatenation tool.

Thinking of going a little further into the weeds.

Instead of denoting "position `a` 8 pts from leading, and distribute `b` and `c` evenly in the remaining space with `.fixed(8) <|> a <|> .relative(1) <|> b <|> .relative(1) <|> c`, I'm thinking `8 <> a <>  ~1~ <> b <> ~1~ <> c`

It's dense, yes. But I think it makes it clear that 8 is a fixed/absolute value, `<>` is "next to" or "anchored", while the 1 is something different (the `~` on either side of a number reminds me of springs).

Note that my operators are limited to the following characters: `/, =, -, +, !, *, %, <, >, &, |, ^, ?, or ~`, and further limited by the following restriction: `The tokens =, ->, //, /*, */, ., the prefix operators <, &, and ?, the infix operator ?, and the postfix operators >, !, and ? are reserved. These tokens can’t be overloaded, nor can they be used as custom operators.` per [Apple’s docs](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/LexicalStructure.html#//apple_ref/doc/uid/TP40014097-CH30-ID418)